traceapp/tmpl/data/trace.html

```diff
@@ -420,6 +428,94 @@
     if(data != null) {
       timelineHover();
     }
+
+    // save state for last sort
+    var lastSort = {
+      name: '',
+      order: false,
+      toggleOrder: function(name){
+        if (this.name !== name) {
+          this.name = name;
+          this.order = false;
+          return 1;
+        }
+
+        this.order = !this.order;
+        if (this.order) {
+          return -1;
+        }
+        return 1;
+      }
+    };
+
+    // sort by starting_time
+    function toggleSortStartTime() {
+      var sortOrder = lastSort.toggleOrder('start_time');
+      data.sort(function(a, b) {
+        // validation
+        if (typeof a.times !== 'object' || typeof b.times !== 'object') return -1;
+        if (a.times.length === 0 || b.times.length === 0) return -1;
+
+        // sort
+        if (a.times[0].starting_time < b.times[0].starting_time) return sortOrder * -1;
+        if (a.times[0].starting_time > b.times[0].starting_time) return sortOrder;
+        return 0
+      })
+      timelineHover();
+    }
+
+    // sort by ending_time
+    function toggleSortEndTime() {
+      var sortOrder = lastSort.toggleOrder('end_time');
+      data.sort(function(a, b) {
+        // validation
+        if (typeof a.times !== 'object' || typeof b.times !== 'object') return -1;
+        if (a.times.length === 0 || b.times.length === 0) return -1;
+
+        // sort
+        if (a.times[0].ending_time < b.times[0].ending_time) return sortOrder * -1;
+        if (a.times[0].ending_time > b.times[0].ending_time) return sortOrder;
```

Comment:
Could be written as an else statement, instead of doing the extra comparison, right?

```
if (a.times[0].ending_time < b.times[0].ending_time) {
    return sortOrder * -1;
} else {
    return 0;
}
```

Also, what is the `sortOrder * -1` needed for?